Welcome Tour: Removed checks for translations

escapemanuele opened the pull request:
#### Changes proposed in this Pull Request

Removed checks used while waiting for translations of the [new slide](https://github.com/Automattic/wp-calypso/pull/59498) introduced in the Welcome Tour.

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

*  Checkout branch
* `yarn dev --sync` from `apps/editing-toolkit` to sync the welcome tour to the sandbox
* Check that the slide "Find your way" (6th on desktop) is correctly showing on desktop
* Change language in [https://wordpress.com/me/account](https://wordpress.com/me/account)
* Check that translations exist

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #59498

escapemanuele committed:
Removed checks for translations

alshakero reviewed (approved):
Thanks for the initiative! LGTM.

alshakero reviewed (changes requested):
It seems tests need some modifications.

escapemanuele committed:
Changed test to check slides

escapemanuele commented:
@alshakero I modified the tests. To be honest, they weren't really helpful, as the locale passed to `getTourSteps` is only used in `createInterpolateElement`( "Congratulations!" slide, so no use in testing "en" and "nl" for "Find your Way" slide!).

While I changed that, I added basic tests for all tests 👍

alshakero reviewed (approved):
Beautiful now!
